src/builder/specification.md

```diff
@@ -0,0 +1,339 @@
+- `attesterSlashings`: `Array`, [`AttesterSlashingV1`](#attesterslashingv1)
+- `attestations`: `Array`, [`AttestationV1`](#attestationv1)
+- `deposits`: `Array`, [`DespositV1`](#depositv1)
+- `voluntaryExits`: `Array`, [`SignedVoluntaryExitV1`](#signedvoluntaryexitv1)
+- `syncAggregate`: `object`, [`SyncAggregateV1`](#syncaggregatev1)
+- `executionPayload`: `object`, [`ExecutionPayloadHeaderV1`](#executionpayloadheaderv1)
+
+### `Eth1DataV1`
+- `depositRoot`: `DATA`, 32 Bytes
+- `depositCount`: `QUANTITY`, 64 Bits
+- `blockHash`: `DATA`, 32 Bytes
+
+### `ProposerSlashingV1`
+- `signedHeader1`: `object`, [`SignedBeaconBlockHeaderV1`](#signedbeaconblockheaderv1)
+- `signedHeader2`: `object`, [`SignedBeaconBlockHeaderV1`](#signedbeaconblockheaderv1)
+
+### `BeaconBlockHeaderV1`
+- `slot`: `QUANTITY`, 64 Bits
+- `proposerIndex`: `QUANTITY`, 64 Bits
+- `parentRoot`: `DATA`, 32 Bytes
+- `stateRoot`: `DATA`, 32 Bytes
+- `bodyRoot`: `DATA`, 32 Bytes
+
+### `AttesterSlashingV1`
+- `attestation1`: `object`, [`IndexedAttestationV1`](#indexedattestationv1)
+- `attestation2`: `object`, [`IndexedAttestationV1`](#indexedattestationv1)
+
+### `IndexedAttestationV1`
+- `attestingIndices`: `Array`, `QUANTITY`, 64 Bits
+- `data`: `object`, [`AttestationDataV1`](#attestationdatav1)
+- `signature`: `DATA`, 96 Bytes
+
+### `AttestationDataV1`
+- `slot`: `QUANTITY`, 64 Bits
+- `index`: `QUANTITY`, 64 Bits
+- `beaconBlockRoot`, `DATA`, 32 Bytes
+- `source`: `object`, [`CheckpointV1`](#checkpointv1)
+- `target`: `object`, [`CheckpointV1`](#checkpointv1)
+
+### `CheckpointV1`
+- `epoch`: `QUANTITY`, 64 Bits
+- `root`: `DATA`, 32 Bytes
+
+### `AttestationV1`
+- `aggregationBits`: `DATA`, 0 to 256 Bytes
+- `data`: `object`, [`AttestationDataV1`](#attestationdatav1)
+- `signature`: `DATA`, 96 Bytes
+
+### `DepositV1`
+- `proof`: `Array`, 32 Bytes
+- `data`: `object`, [`DepositDataV1`](#depositdatav1)
+
+### `DepositDataV1`
+- `pubkey`: `DATA`, 48 Bytes
+- `withdrawalCredentials`: `DATA`, 32 Bytes
+- `amount`: `QUANTITY`, 64 Bits
+- `signature`: `DATA`, 96 Bytes
+
+### `VoluntaryExitV1`
+- `epoch`: `QUANTITY`, 64 Bits
+- `ValidatorIndex`: `QUANTITY`, 64 Bits
+
+### `SyncAggregateV1`
+- `syncCommitteeBits`: `DATA`, 64 Bytes
+- `syncCommitteeSignature`: `DATA`, 96 Bytes
+
+#### Signed Containers
+
+### `SignedBeaconBlockHeaderV1`
+- `message`: `object`, [`BeaconBlockHeader`](#beaconblockheaderv1)
+- `signature`: `DATA`, 96 Bytes
+
+### `SignedVoluntaryExitV1`
+- `message`: `object`, [`VoluntaryExitV1`](#voluntaryexitv1)
+- `signature`: `DATA`, 96 Bytes
+
+#### SSZ Objects
+
+Consider the following definitions supplementary to the definitions in [`consensus-specs`][consensus-specs].
+
+##### `ValidatorRegistrationV1`
+
+```python
+class ValidatorRegistrationV1(Container):
+    feeRecipient: Bytes20
+    timestamp: uint64
+    pubkey: BLSPubkey
+```
+
+##### `BuilderReceiptV1`
+
+```python
+class BuilderReceiptV1(Container):
```

Comment:
name change?

```suggestion
class BuilderBidV1(Container):
```

I think "bids" is more appropriate here than "receipts" (which implies the output of a finished process) and this is in the middle of what could become a very dynamic process

this terminology is also forwards compatible with any PBS stuff we do down the line